lib/use-display-currency.tsx

```diff
@@ -0,0 +1,310 @@
+import { gql } from "@apollo/client"
+import {
+  useCurrencyListQuery,
+  useRealtimePriceQuery,
+  WalletCurrency,
+} from "./graphql/generated"
+import {
+  ConvertMoneyAmount,
+  DisplayAmount,
+  DisplayCurrency,
+  lessThan,
+  MoneyAmount,
+  WalletAmount,
+  WalletOrDisplayCurrency,
+} from "./types/amounts"
+import { useCallback, useMemo } from "react"
+import { usePriceConversion } from "./use-price-conversion"
+
+gql`
+  query currencyList {
+    currencyList {
+      __typename
+      id
+      flag
+      name
+      symbol
+      fractionDigits
+    }
+  }
+`
+
+const usdDisplayCurrency = {
+  symbol: "$",
+  id: "USD",
+  fractionDigits: 2,
+}
+
+const defaultDisplayCurrency = usdDisplayCurrency
+
+const formatCurrencyHelper = ({
+  amountInMajorUnits,
+  symbol,
+  fractionDigits,
+  withSign = true,
+  withDecimals = true,
+}: {
+  amountInMajorUnits: number | string
+  symbol: string
+  fractionDigits: number
+  withSign?: boolean
+  withDecimals?: boolean
+}) => {
+  const isNegative = Number(amountInMajorUnits) < 0
+  const decimalPlaces = withDecimals ? fractionDigits : 0
+  const amountStr = Intl.NumberFormat("en-US", {
+    minimumFractionDigits: decimalPlaces,
+    maximumFractionDigits: decimalPlaces,
+    // FIXME this workaround of using .format and not .formatNumber is
+    // because hermes haven't fully implemented Intl.NumberFormat yet
+  }).format(Math.abs(Number(amountInMajorUnits)))
+  return `${isNegative && withSign ? "-" : ""}${symbol}${amountStr}`
+}
+
+const displayCurrencyHasSignificantMinorUnits = ({
+  convertMoneyAmount,
+  amountInMajorUnitOrSatsToMoneyAmount,
+}: {
+  convertMoneyAmount?: ConvertMoneyAmount
+  amountInMajorUnitOrSatsToMoneyAmount: (
+    amount: number,
+    currency: WalletOrDisplayCurrency,
+  ) => MoneyAmount<WalletOrDisplayCurrency>
+}) => {
+  if (!convertMoneyAmount) {
+    return true
+  }
+
+  const oneMajorUnitOfDisplayCurrency = amountInMajorUnitOrSatsToMoneyAmount(
+    1,
+    DisplayCurrency,
+  )
+
+  const oneUsdCentInDisplayCurrency = convertMoneyAmount(
+    {
+      amount: 1,
+      currency: WalletCurrency.Usd,
+    },
+    DisplayCurrency,
+  )
+
+  return lessThan({
+    value: oneUsdCentInDisplayCurrency,
+    lessThan: oneMajorUnitOfDisplayCurrency,
+  })
+}
+
+export const useDisplayCurrency = () => {
```

Comment:
I think there is a decent amount of code in this that is unused that can be removed. I believe it will also allow you to remove the `use-price-conversion` file.

Reply:
ah yes, i just copied alot of that code from the mobile app and ended up doing it another way based on how the state was already being managed in the POS

Reply:
removed unused code